algorithms/kernel/svm/svm_train_thunder_impl.i

```diff
@@ -101,25 +100,60 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, ParameterType, cpu>::com
     DAAL_CHECK_MALLOC(yTArray.get());
     algorithmFPType * const y = yTArray.get();
 
+    SafeStatus safeStat;
+
+    size_t nNonZeroWeights = nVectors;
     {
-        ReadColumns<algorithmFPType, cpu> mtY(yTable, 0, 0, nVectors);
-        DAAL_CHECK_BLOCK_STATUS(mtY);
-        const algorithmFPType * const yIn = mtY.get();
-
-        ReadColumns<algorithmFPType, cpu> mtW(wTable.get(), 0, 0, nVectors);
-        DAAL_CHECK_BLOCK_STATUS(mtW);
-        const algorithmFPType * weights = mtW.get();
-        // gradi = -yi; ai = 0; ci = c*wi
-        for (size_t i = 0; i < nVectors; i++)
+        const size_t blockSize = 16384;
+        const size_t nBlocks   = nVectors / blockSize + !!(nVectors % blockSize);
+
+        TArrayScalable<size_t, cpu> weightsCounter;
+        if (wTable.get())
+        {
+            weightsCounter.reset(blockSize);
+            DAAL_CHECK_MALLOC(weightsCounter.get());
+        }
+
+        daal::threader_for(nBlocks, nBlocks, [&](const size_t iBlock) {
+            const size_t startRow     = iBlock * blockSize;
+            const size_t nRowsInBlock = (iBlock != nBlocks - 1) ? blockSize : nVectors - iBlock * blockSize;
+
+            ReadColumns<algorithmFPType, cpu> mtY(yTable, 0, startRow, nRowsInBlock);
+            DAAL_CHECK_BLOCK_STATUS_THR(mtY);
+            const algorithmFPType * const yIn = mtY.get();
+
+            ReadColumns<algorithmFPType, cpu> mtW(wTable.get(), 0, startRow, nRowsInBlock);
+            DAAL_CHECK_BLOCK_STATUS_THR(mtW);
+            const algorithmFPType * weights = mtW.get();
+
+            if (weights)
+            {
+                weightsCounter[iBlock] = 0;
+            }
+            for (size_t i = 0; i < nRowsInBlock; ++i)
+            {
+                y[i + startRow]     = yIn[i] == 0 ? algorithmFPType(-1) : yIn[i];
+                grad[i + startRow]  = -y[i + startRow];
+                alpha[i + startRow] = algorithmFPType(0);
+                cw[i + startRow]    = weights ? weights[i] * C : C;
+                if (weights)
+                {
+                    weightsCounter[iBlock] += static_cast<size_t>(weights[i] != algorithmFPType(0));
```

Comment:
minor: be shure that it's not a problem for threading cache false sharing, may be it would be better to store accumulated variables into tls and then reduce it